### PR TITLE
Only enforce policy on AWS managed keys

### PR DESCRIPTION
**Pull request by bseb**
### Background
As mentioned [here](https://docs.aws.amazon.com/kms/latest/developerguide/rotate-keys.html) any non AWS managed keys are not eligible for automatic rotation, we should not try to enforce a rotation policy on those

### Changes
Short circut if the Origin is not `AWS_KMS`

### Testing
Unit and CI pipeline

### CODE REVIEW

**Review by lindsey-w — APPROVED** [review]
🥳

File: aws_kms_policies/aws_cmk_key_rotation.yml [hunk 1]
```diff
@@ -113,3 +113,26 @@ Tests:
         "ValidTo": null,
         "Policy": "{JSON policy document see https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html for details}"
       }
+  -
+    Name: Key Not Eligible for Roration
```
Comment (lindsey-w) [inline aws_kms_policies/aws_cmk_key_rotation.yml:117]:
```suggestion
    Name: Key Not Eligible for Rotation
```

**Review by darwayne — DISMISSED** [review]

File: aws_kms_policies/aws_cmk_key_rotation.yml (hunk 1 above)
Comment (bseb) [reply aws_kms_policies/aws_cmk_key_rotation.yml:117]: Typing is hard

File: aws_kms_policies/aws_cmk_key_rotation.yml [hunk 2]
```diff
@@ -113,3 +113,26 @@ Tests:
...
+    ExpectedResult: true
+    Resource:
+      {
+        "KeyRotationEnabled": true,
```
Comment (lindsey-w) [inline aws_kms_policies/aws_cmk_key_rotation.yml:121]: make this false or null just to ensure it is still ignored?

Comment (bseb) [reply aws_kms_policies/aws_cmk_key_rotation.yml]: changed to null, still looks good 🎉

**Review by lindsey-w — APPROVED** [review]

**Review by darwayne — APPROVED** [review]

**Review by wey-chiang — APPROVED** [review]

**Pull request merged**